### ISSUE TITLE
Spotifylyrics.exe opens and closes immediately on Windows 10

### ISSUE THREAD

**Issue by EvilLink** [SimonIT/spotifylyrics#21]
I use Windows 10 Version 1909, when I open the app it closes immediately, it shows in task manager list but vanishes. There's no error message or anything else.

**Comment by SimonIT** [comment]
Do you tried to install _Visual C++ Redistributable for Visual Studio 2015_ as mentioned in the readme?

**Comment by EvilLink** [comment]
Yeah, I have it installed.

**Comment by SimonIT** [comment]
Then I have no clue why it doesn't work. By doing a quick research I couldn't find any problems with pyinstaller on windows.
Alternatively you can install python and clone the repository.

**Comment by EvilLink** [comment]
Now I've tried the 1.40 version and it opened normally... But when I try to open the 1.50, happens what I said before...

**Comment by SimonIT** [comment]
Hmm, the only difference is that I generated 1.40 on my own computer and 1.50 on travis which is a Windows Server, version 1809

**Comment by kleytondread** [comment]
I would create one ticket but my problem is similar to EvilLink, except that mine doesn't even start the .exe file in version 1.50, though it executes the 1.40 version normally (actually there is a problem with version 1.40 -- the reason why I tried the update -- it takes a good 30-50 seconds to start the application). So I don't know what could be the cause either, since last version runs it but the latest does not.

**Comment by SimonIT** [comment]
So strange.
Can you both write some Details about your Windows version? Maybe we can find the problem this way

**Comment by kleytondread** [comment]
@SimonIT here is my 'about' on the windows version, sorry, it is all in pt-BR, if you don't understand something just let me know.
Also if you need some other detail about my windows let me know, I'll be happy to help you.

![Capturar](https://user-images.githubusercontent.com/11301984/80140742-c7a3dd80-857e-11ea-90f7-be13e33bd605.JPG)

**Comment by kleytondread** [comment]
ps: I noticed my windows version is not updated. i'm trying to update now, and I'll report back to you once I install it and reboot my pc.ç

**Comment by SimonIT** [comment]
Hm, the only differences to mine one are a newer version (I got version 1909 and build 18363.815) and an Intel processor. I personally don't think that these two differences matters.

Is there an error message if you start the .exe file from cmd?

**Comment by kleytondread** [comment]
Ok, I updated my W10 version to 1909. Downloaded the 1.50 .exe file again, and it has the same problem, version 1.50 doesn't even open. No signs of its process on the Task manager. Version 1.40 still works, though it takes 30sec or so to open.

New W10 version:
![image](https://user-images.githubusercontent.com/11301984/80142457-8eb93800-8581-11ea-8338-b2f30f13d257.png)

Running through CMD:
Version 1.40 - opens as usual;
Version 1.50 - nothing so far, and no warning massage. Here's a screen capture with both icons os the app and also showing the 1.40 opened... no 1.50 though.

![image](https://user-images.githubusercontent.com/11301984/80143140-bbba1a80-8582-11ea-9b87-da5ab10ca647.png)

**Comment by SimonIT** [comment]
I tried now in a fresh installed Windows VM and it worked fine. I have no clue why it doesn't work for you both. If you got ideas, please share or try them. I would be very happy if we even get an error message.
I can make the exe on my own pc like I did for 1.4, but as you mentioned it has a very long delay at the start. I don't know if it's worth. I will set the log level to debug at generating to get hopefully some output.

**Comment by kleytondread** [comment]
@SimonIT does the Version 1.50 generates any log file or AppData folder? I tried finding it, but no sings of it.

**Comment by SimonIT** [comment]
There isn't any logging implemented, only some console outputs. Maybe it's now time implement something in this direction

**Comment by kleytondread** [comment]
I'll keep you posted in case I find a way to make it work, though I don't know what else to do in order to teste it. But thank you very very much for the time and the updates!

**Comment by EvilLink** [comment]
After all this time, I was thinking here... @kleytondread and me are from Brazil. Could it be possible any settings of region, location or format that our windows use and the Python can't access correctly to load? I already experienced something like this in others apps

**Comment by SimonIT** [comment]
Hmm, that could be possible. Have you special characters in your usernames? I know that Java has sometimes problems with that, so I can think of having a problem with this here, too.

**Comment by kleytondread** [comment]
If you mean spotify username, than no, no special characters. But I do use facebook authentication on my spotify, if that helps in anyway shape or form.

**Comment by SimonIT** [comment]
No, I mean you windows username

**Comment by kleytondread** [comment]
also no.

**Comment by peter9811** [comment]
W10 Pro, x64, 1909
![image](https://user-images.githubusercontent.com/22783445/85079271-0378af00-b18c-11ea-8e37-8551430215df.png)

**Comment by SimonIT** [comment]
Okay guys, I just released 1.52 with error reporting to sentry. If you update I will hopefully get an error message indicating what's going wrong

**Comment by EvilLink** [comment]
Same problem... I made a GIF:

![2020-06-19_06-41-16](https://user-images.githubusercontent.com/47981801/85119484-35275f80-b1f8-11ea-9080-e1aa6aa04671.gif)

**Comment by EvilLink** [comment]
Did you implemented a debug log mode or something like that, that I can do through CMD?

**Comment by SimonIT** [comment]
No, I included sentry. But there appears no error. Now I let travis build the exe file without windowed mode, so there is a console which should show the error. https://github.com/SimonIT/spotifylyrics/releases/tag/1.52-DEBUG

**Comment by EvilLink** [comment]
I tried to open through CMD or clicking on it. Same problem yet.

**Comment by SimonIT** [comment]
Can you redownload https://github.com/SimonIT/spotifylyrics/releases/tag/1.52-DEBUG ? I made sure that now a console window opens with a log

**Comment by EvilLink** [comment]
Now it opened! 

![chrome_2020-06-19_08-37-35](https://user-images.githubusercontent.com/47981801/85128857-7b84ba80-b208-11ea-863d-7a225f0cfeb0.jpg)

![chrome_2020-06-19_08-39-10](https://user-images.githubusercontent.com/47981801/85128864-7f184180-b208-11ea-874f-6606ec7c5bae.jpg)

**Comment by SimonIT** [comment]
But how????? I mean I didn't change anything despite except .pyw to .py and removed the windowed flag from the pyinstaller command

**Comment by EvilLink** [comment]
I have no idea how, but now it works normally with this log window.... It must be ALIENS! :P

**Comment by EvilLink** [comment]
I'm not Python programmer but, I googled a little what you said and I found some people complaining about .pyw and .py, it appears to be the problem.

https://stackoverflow.com/questions/21545667/cant-execute-pyw-on-windows

**Comment by SimonIT** [comment]
I think we have to try it without pyw but with -w, maybe that works

**Comment by EvilLink** [comment]
Let's do that, what I have to?

**Comment by kleytondread** [comment]
Hey guys, @SimonIT what file should I download to test it (the .exe 1.52)?

**Comment by SimonIT** [comment]
@kleytondread The exe from https://github.com/SimonIT/spotifylyrics/releases/tag/1.52-DEBUG seem to work for EvilLink

https://github.com/SimonIT/spotifylyrics/releases/tag/1.52-DEBUG2 Here is the exe with .py but with the windowed flag for pyinstaller. If this works, I would be very happy

**Comment by EvilLink** [comment]
LOL... Now it doesn't work..... Only the previous debug...

**Comment by SimonIT** [comment]
Ahhhh, I'm freaking out. It could be this problem: pyinstaller/pyinstaller#4930, but I'm absolute unsure

**Comment by EvilLink** [comment]
Well... I'll use the debug version while that... Chill your mind and let's forget this for a while, don't freak out

**Comment by kleytondread** [comment]
@SimonIT, I can confirm this Debug version (1.52) opens fast, like it should. i'll try downloading the non-debug version to check it

**Comment by EvilLink** [comment]
I was opening from Python console until 1.53, but now the 1.54 '.exe' opens normally (again)... Just to let you know... LOL

**Comment by SimonIT** [comment]
Crazy

**Comment by SimonIT** [comment]
Hey guys, I changed the building from travis to github actions. https://github.com/SimonIT/spotifylyrics/suites/1524211646/artifacts/26916651 Here is the newest exe. Maybe this does work

**Comment by kleytondread** [comment]
Hey, SimonIT, sorry for the long time to reply. só far I have tested this new build in one of my computers (version has been stable and it opens properly). I cannot verify, however, on my desktop -- which is the one I was having problems with -- not until January, when I'm back home. With that said I'm gonna start studying your code, I feel I can learn a lot with it, I'm don't know much about Python, I only worked with Java and C# so far. But I wanna see if I can learn enough so I can try helping you (if you allow me, of course).

_PS: I'll try replaying this again once I have access to my Desktop_
